### PR TITLE
semgrep: fix build and bump to 1.37.0

### PR DESCRIPTION
**Pull request by brandonspark**
This PR should fix the build error for the `semgrep` homebrew package, which is due to using a `Makefile` target which no longer exists. See https://github.com/returntocorp/semgrep/pull/8512

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/semgrep.rb [hunk 1]
```diff
@@ -211,7 +211,6 @@ def install
       # Install semgrep-core and spacegrep
       system "opam", "install", "--deps-only", "-y", "."
       system "opam", "exec", "--", "make", "core"
-      system "opam", "exec", "--", "make", "core-install"
```
Comment (ZhongRuoyu) [inline Formula/s/semgrep.rb:214]: It's likely still needed for now as changes in https://github.com/returntocorp/semgrep/pull/8512 are not included in a release.

Comment (chenrui333) [reply Formula/s/semgrep.rb:214]: yeah, upstream first, let's release on the semgrep side and fix along the way

**Comment by brandonspark** [comment]
@chenrui333 @ZhongRuoyu it should be good now, the most recent Semgrep release has landed

**Comment by ZhongRuoyu** [comment]
@brandonspark it still needs #140451. Can you add those changes to this PR? We can close #140451 then.

**Comment by brandonspark** [comment]
@ZhongRuoyu OK, is this correct?

**Comment by ZhongRuoyu** [comment]
Yes. Thanks, @brandonspark.

**Comment by ZhongRuoyu** [comment]
Seems that `semgrep-core` couldn't be found for some reason:

```
  ==> Testing semgrep
  ==> /opt/homebrew/Cellar/semgrep/1.37.0/bin/semgrep --help
  Failed to find semgrep-core in PATH or in the semgrep package.
  Error: semgrep: failed
```

**Comment by ZhongRuoyu** [comment]
CI is quite busy right now, so let's try without https://github.com/Homebrew/homebrew-core/labels/long%20build first.

**Comment by brandonspark** [comment]
@ZhongRuoyu looks like its ok now!

**Review by ZhongRuoyu — DISMISSED** [review]
Thanks @brandonspark!

**Pull request closed without merging**